src/bindings/python/src/pyopenvino/graph/dict_attribute_visitor.cpp

```diff
@@ -274,6 +274,18 @@ void util::DictAttributeSerializer::on_adapter(const std::string& name, ov::Valu
     if (m_attributes.contains(name)) {
         OPENVINO_THROW("No AttributeVisitor support for accessing attribute named: ", name);
     }
+    
+    if (auto _adapter = dynamic_cast<ov::AttributeAdapter<std::shared_ptr<ov::op::util::Variable>>*>(&adapter)) {
```

Comment:
May be it'd be better to add several overloads instead of `if-else`?
@jiwaszki what do you think?